src/main/java/com/feywild/feywild/block/MandrakeCrop.java

```diff
@@ -109,8 +110,12 @@ public InteractionResult use(@Nonnull BlockState state, @Nonnull Level level, @N
                     if (!player.isCreative()) player.getItemInHand(hand).shrink(1);
                     QuestData.get((ServerPlayer) player).checkComplete(SpecialTask.INSTANCE, SpecialTaskAction.SUMMON_MANDRAGORA);
                 }
+            } else if (!level.isClientSide && QuestData.get((ServerPlayer) player).getAlignment() != Alignment.SPRING) {
```

Comment:
Unnecessary complicated condition. The whole `if` could be inside another `if` that checks whether the level is client side. And then you could use a simple `if` - `else`.

Also it might be worth adding a `checkReputation` method to `QuestData` that takes an `Alignment and a minimum reputation and checks whether the player has that alignment and at least the amount of reqputation.